File: src/copilot_console/app/models/pin.py
"""Pin models."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class Pin(BaseModel):
    """A durable per-session pin anchored to an SDK message id."""

    id: str
    session_id: str
    sdk_message_id: str
    created_at: datetime
    updated_at: datetime
    title: str | None = None
    excerpt: str | None = None
    note: str | None = None
    tags: list[str] | None = None


class PinCreate(BaseModel):
    """Request to create a pin."""

    sdk_message_id: str = Field(..., min_length=1)
    title: str | None = None
    excerpt: str | None = None
    note: str | None = None
    tags: list[str] | None = None


class PinUpdate(BaseModel):
    """Request to update mutable pin fields."""

    title: str | None = None
    excerpt: str | None = None
    note: str | None = None
    tags: list[str] | None = None

File: frontend/src/types/pin.ts
export interface Pin {
  id: string;
  session_id: string;
  sdk_message_id: string;
  created_at: string;
  updated_at: string;
  title?: string | null;
  excerpt?: string | null;
  note?: string | null;
  tags?: string[] | null;
}

export interface CreatePinRequest {
  sdk_message_id: string;
  title?: string;
  excerpt?: string;
  note?: string;
  tags?: string[];
}

export interface UpdatePinRequest {
  title?: string;
  excerpt?: string;
  note?: string;
  tags?: string[];
}
